### PR TITLE
Crimbo Train (Locomotive), The Superconductor, and CPU

### PR DESCRIPTION
**Pull request by Veracity0**
1) Crimbo Train (Locomotive)
2) The Superconductor is a monster.
Once defeated, superconductorDefeated is true
3) The Superconductor's CPU has modifiers affected by elfGratitude
4) two more tiny plastic advent calendar items.

### CODE REVIEW

File: src/data/modifiers.txt [hunk 1]
```diff
@@ -3621,6 +3621,8 @@ Item	the Ley Incursion's waist	Mysticality: +15, Mysticality Percent: +25, Exper
 Item	The Lost Glasses	Reduce Enemy Defense: 15, Hot Resistance: +2, Single Equip
 Item	The One Mood Ring	Muscle Percent: [5*G], Mysticality Percent: [5*G], Moxie Percent: [5*G], Monster Level: [25-5*G], Single Equip
 Item	The Ring	Muscle: +3, Mysticality: +3, Moxie: +3, Spooky Damage: +40, Single Equip
+# Item The Superconductor's CPU: (based on Crimbo 22 Elf Gratitude)
+Item	The Superconductor's CPU	Maximum HP: [floor(sqrt(pref(elfGratitude)))], Maximum MP: [floor(sqrt(pref(elfGratitude)))]
```
Comment (midgleyc) [inline src/data/modifiers.txt:3625]: Does this cause issues if the gratitude pref has commas in it?

File: src/data/adventures.txt [hunk 2]
```diff
@@ -414,6 +414,7 @@ Crimbo22	adventure=559	DiffLevel: mid Env: indoor Stat: 0 nowander	Crimbo Train
 Crimbo22	adventure=560	DiffLevel: mid Env: indoor Stat: 0 nowander	Crimbo Train (Passenger Car)
 Crimbo22	adventure=561	DiffLevel: mid Env: indoor Stat: 0 nowander	Crimbo Train (Dining Car)
 Crimbo22	adventure=562	DiffLevel: mid Env: indoor Stat: 0 nowander	Crimbo Train (Coal Car)
+Crimbo22	place=crimbo22_engine	DiffLevel: mid Env: indoor Stat: 0 nowander	Crimbo Train (Locomotive)
```
Comment (midgleyc) [inline src/data/adventures.txt:417]: as a "place", I'm not sure if it has an environment. and it probably doesn't appear for autumnaton either, so we can't know the difflevel. Still, this is probably fine.

File: src/net/sourceforge/kolmafia/session/QuestManager.java [hunk 3]
```diff
@@ -2080,6 +2080,8 @@ else if (responseText.contains("Dozens of nearby smut orcs")) {
       QuestDatabase.setQuest(Quest.MAELSTROM, QuestDatabase.UNSTARTED);
 
       Preferences.setInteger("lastThingWithNoNameDefeated", KoLCharacter.getAscensions());
+    } else if (monsterName.equals("The SuperConductor")) {
```
Comment (midgleyc) [inline src/net/sourceforge/kolmafia/session/QuestManager.java:2083]: just `The Superconductor`, not `The SuperConductor`?

also in-game it starts with a space (` The Superconductor`), do we trim monster names already?

File: src/data/modifiers.txt (hunk 1 above)
Comment (Veracity0) [reply src/data/modifiers.txt:3625]: It does not.

All Preferences are strings.
Preferences.getInteger("elfGratitude") passes it through StringUtilities.parseInteger, which handles commas.
So does ash get_property("elfGratitude").to_integer()

File: src/data/adventures.txt (hunk 2 above)
Comment (Veracity0) [reply src/data/adventures.txt:417]: Ah. Yeah, It was copy/paste/modify and I didn't even think about it.

If there is another comment which requires an edit, I will fix it.

File: src/net/sourceforge/kolmafia/session/QuestManager.java (hunk 3 above)
Comment (Veracity0) [reply src/net/sourceforge/kolmafia/session/QuestManager.java:2083]: We do.

File: src/data/modifiers.txt (hunk 1 above)
Comment (Veracity0) [reply src/data/modifiers.txt:3625]:
```
> get elfGratitude

2,335

> ash string_modifier($item[The Superconductor's CPU], "Evaluated Modifiers")

Returned: Maximum HP: +48, Maximum MP: +48
```

**Review by midgleyc — APPROVED** [review]
Thanks!

**Pull request merged**